### PR TITLE
Add to_unoptimized_plan

### LINKED ISSUE #3340: Add ability to get unoptimized logical plan from DataFrame
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
It is only possible to get an optimized logical plan from a DataFrame (by calling `to_logical_plan`).

**Describe the solution you'd like**
Add ability to get unoptimized logical plan from DataFrame

**Describe alternatives you've considered**
None

**Additional context**
None

### PR DESCRIPTION
**Pull request by iajoiner**
# Which issue does this PR close?

Closes #3340.

 # Rationale for this change

Enable people to fetch the raw `LogicalPlan` in a `DataFrame`.
# What changes are included in this PR?

`pub fn to_raw_logical_plan(&self) -> LogicalPlan {..} `has been added together with an assertion in a test.
# Are there any user-facing changes?

Yes. There are no breaking changes though.

### CODE REVIEW

**Comment by iajoiner** [comment]
cc @andygrove

File: datafusion/core/src/dataframe.rs [hunk 1]
```diff
@@ -520,7 +520,13 @@ impl DataFrame {
         self.plan.schema()
     }
 
-    /// Return the logical plan represented by this DataFrame.
+    /// Return the raw (i.e. unoptimized) logical plan represented by this DataFrame.
+    pub fn to_raw_logical_plan(&self) -> LogicalPlan {
```
Comment (andygrove) [inline datafusion/core/src/dataframe.rs:524]: I wonder if a better name would be `to_unoptimized_plan`?

Comment (iajoiner) [reply datafusion/core/src/dataframe.rs:524]: Sure. Do we need to specify that it is a logical as opposed to physical plan in the name?

Comment (andygrove) [reply datafusion/core/src/dataframe.rs]: I don't think so, because the return type is `LogicalPlan`

**Review by andygrove — APPROVED** [review]
LGTM. Thanks @iajoiner!

File: datafusion/core/src/dataframe.rs (hunk 1 above)
Comment (iajoiner) [reply datafusion/core/src/dataframe.rs]: Sure!

Comment (iajoiner) [reply datafusion/core/src/dataframe.rs]: @andygrove  Done!

**Pull request merged**